release/tar/linux/opensearch-tar-build.sh

```diff
@@ -0,0 +1,175 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# About:        This shell script generate an opensearch tarball with all the provided plugins installed and ready to use. 
+#               A manifest file and right permissions set up is essential to download the required artifacts
+# Dependencies: yq (More info: https://github.com/mikefarah/yq/tree/v4.4.1#install)
+# Usage: .      /opensearch-tar-build.sh -f <path/to/manifest/file/>
+#
+
+set -e
+
+function usage() {
+    echo "Usage: $0 [args]"
+    echo -e "Supported manifest schema-version: 1.0"
+    echo ""
+    echo "Required arguments:"
+    echo -e "-f MANIFEST_FILE\tPath/to/manifest.yml file"
+    echo ""
+    echo "Optional arguments:"
+    echo -e "-p PLATFORM\tSpecify the platform. eg: linux, macos, windows. Currently only supports and defaults to linux"
+    echo -e "-a ARCHITECTURE\tSpecify the architecture, e.g. x64, arm64. Currently only supports x64 and arm64. Defaults to x64"
+    echo -e "-h help"
+}
+
+while getopts ":ha:p:f:" arg; do
+    case $arg in
+        h)
+            usage
+            exit 1
+            ;;
+        p)
+            PLATFORM=$(echo "$OPTARG" | tr '[:upper:]' '[:lower:]')
+            ;;
+        a)
+            ARCHITECTURE=$(echo "$OPTARG" | tr '[:upper:]' '[:lower:]')
+            ;;
+        f)
+            MANIFEST_FILE=$OPTARG
+            ;;
+        :)
+            echo "Error: -${OPTARG} requires an argument"
+            usage
+            exit 1
+            ;;
+        ?)
+            echo "Invalid option: -${arg}"
+            exit 1
+            ;;
+    esac
+done
+
+if [ -z "$PLATFORM" ]; then
+    PLATFORM="linux"
+fi
+
+if [ "$PLATFORM" != "linux" ]; then
+    echo "error: $PLATFORM platform is not supported yet!"
+    usage
+    exit 1
+fi
+
+if [ -z "$ARCHITECTURE" ]; then
```

Comment:
Architecture should be required or it will error out - avoids being unclear about what is produced

Reply:
Offline recap:
What if we moved these parameters into the manifest itself?

> As the artifacts listed are platform specific it doesn't make much sense to change the way you produce the product to be `./opensearch-tar-build.sh -f manifest.yml -a x64` and then do `./opensearch-tar-build.sh -f manifest.yml -a arm64` and you get the exact same but bundled up in an incorrectly named bundle output.

This would effective make the only parameter to this script be the manifest file.  Then it might make sense as convention to suggestion `manifest.linux.x64.yml` which is duplicated in the lines of the manifest file

Reply:
+1

Reply:
How does this look?

```
manifest:
  schema-version: 1.0
  platform: linux
  architecture: x64
```

Reply:
I would be inclined to make platform part of the product (where a product is OpenSearch) as there are components that are interpreted and can run on any platform, so it isn't the manifest that has these properties but the produced artifact that does.

```
manifest:
   schema-version: 1.0
   description: Creates the OpenSearch bundle of core engine and default plugins
   product:
      name: OpenSearch
      platform: linux
      architecture: x64
```

Added some extra fields that make the manifest a little more verbose.  That prose isn't used by the script, but it could be useful if you've got a big stack of these manifest and you want some historical context

Reply:
Platform and architecture don't belong in the `manifest` section, because they're not describing the manifest itself. I think they should be under each `product`. That will allow us to have one manifest that can build all four products (core, dashboards, core-arm, dashboards-arm) with the same product version number:

```
products:
  OpenSearch-linux-x64:
    platform: linux
    architecture: x64
    ...
  OpenSearch-linux-arm64:
    platform: linux
    architecture: arm64
    ...
```

Alternatively, a separate `build` section might make sense:

```
manifest:
  schema-version: 1.0
build:
  version: 1.0.0
  platform: linux
  architecture: x64
products:
  ...
```

The first option gives us build-the-whole-world-with-one-manifest, the second option gives us simpler manifests with no need for any logic about how to generate output filenames. Now that I write that sentence down, I think option 2 makes the most sense. Keep it simple.

Reply:
Right! It would be better to NOT hard code the key (eg: OpenSearch-linux-x64). Rather keep it in generalized. Will go with option 2 for now. Any future enhancements can be a part of manifest schema-version `2.0`

Reply:
done